### PR TITLE
RavenDB-16968 - Avoid calling DiscardVirtualMemory on Windows by default

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-16968

### CODE REVIEW

File: src/Voron/Impl/Paging/AbstractPager.cs [hunk 1]
```diff
@@ -26,7 +23,7 @@ namespace Voron.Impl.Paging
     public abstract unsafe class AbstractPager : IDisposable, ILowMemoryHandler
     {
         public readonly Logger Log = LoggingSource.Instance.GetLogger<AbstractPager>("AbstractPager");
-        private readonly StorageEnvironmentOptions _options;
+        public readonly StorageEnvironmentOptions _options;
```
Comment (ayende) [inline src/Voron/Impl/Paging/AbstractPager.cs]: Revert this, please

Comment (grisha-kotler) [reply src/Voron/Impl/Paging/AbstractPager.cs]: done

File: src/Voron/StorageEnvironmentOptions.cs [hunk 2]
```diff
@@ -1233,7 +1233,8 @@ public int TimeToSyncAfterFlushInSec
         public bool SkipChecksumValidationOnDatabaseLoading { get; set; }
 
         public int MaxNumberOfRecyclableJournals { get; set; } = 32;
-
+        public bool DiscardVirtualMemory { get; set; }
```
Comment (ppekrol) [inline src/Voron/StorageEnvironmentOptions.cs]: Maybe set default to true? Some tests are creating StorageEnv directly. Should not matter much, but I think its better to have it turn on.

Comment (grisha-kotler) [reply src/Voron/StorageEnvironmentOptions.cs]: done

**Review by arekpalinski — COMMENTED** [review]
Asked some questions in the issue https://issues.hibernatingrhinos.com/issue/RavenDB-16968

File: src/Raven.Server/Config/Categories/StorageConfiguration.cs [hunk 3]
```diff
@@ -2,11 +2,24 @@
...
+using Sparrow.Platform;
 
 namespace Raven.Server.Config.Categories
 {
     public class StorageConfiguration : ConfigurationCategory
     {
+        public StorageConfiguration()
+        {
+            // On Windows, DiscardVirtualMemory can result in high CPU usage due to contention on the
+            // PTE entry (in Win Server 2016 and Win Server 2019), we want to avoid it by default.
```
Comment (arekpalinski) [inline src/Raven.Server/Config/Categories/StorageConfiguration.cs:14]: This is specific to WS2016 and WS2019, right? But we're disabling it on all Windows versions. Is it really what we want?

Comment (ayende) [reply src/Raven.Server/Config/Categories/StorageConfiguration.cs:14]: We saw that in those two, and we don't _need_ to have it. Better to have more consistent behavior than try to test across everything that exists.

**Pull request merged**